src/main/java/scala_maven/ScalaMojoSupport.java

```diff
@@ -783,47 +789,89 @@ private String getToolClasspath() throws Exception {
     return options;
   }
 
+  private String getScala3ArtifactId(String a) {
+    String _scalaVersionNStr = _scalaVersionN.toString();
+    // int indexOf_ = _scalaVersionNStr.indexOf("-");
+    return _scalaVersionN.major == 3 ? a + "_" + _scalaVersionNStr : a;
```

Comment:
This will not work for Scala 3.0.0 onward because the binary version will be "3".
```scala
         val binaryVersion =
            if(scalaVersionN.major == 3 && scalaVersionN.minor == 0 && scalaVersionN.bugfix == 0 && scalaVersionN.modifier != null) scalaVersionN.toString
           else "3"
```

`binaryVersion` should be `"3.0.0-x"` if Scala version is 3.0.0-x or 3.0.0-x-bin-y
but it should be `"3"` if Scala version >= 3.0.0  (for instance 3.0.0, 3.0.1-RC1, 3.0.1, 3.1.0...)